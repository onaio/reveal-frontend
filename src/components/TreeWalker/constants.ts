export const ACTIVE = 'Active';
export const FEATURE = 'Feature';
export const FIND_BY_ID = 'location/findByJurisdictionIds';
export const FIND_BY_PROPERTIES = 'location/findByProperties';
export const LOCATION = 'location';
export const COULDNT_LOAD_CHILDREN = 'Could not load children';
export const COULDNT_LOAD_PARENTS = 'Could not load parents';
export const NOT_AVAILABLE = 'N/A';
