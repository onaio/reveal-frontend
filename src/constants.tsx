// strings
export const MAP = 'map';
export type MAP = typeof MAP;
export const LOCATION = 'Location';
export type LOCATION = typeof LOCATION;
export const PROVINCE = 'Province';
export type PROVINCE = typeof PROVINCE;
export const HOME = 'Home';
export type HOME = typeof HOME;
export const FOCUS_INVESTIGATION = 'Focus Investigation';
export type FOCUS_INVESTIGATION = typeof FOCUS_INVESTIGATION;
export const ACTIVE_FOCUS_INVESTIGATION = 'Active Focus Investigations';
export type ACTIVE_FOCUS_INVESTIGATION = typeof ACTIVE_FOCUS_INVESTIGATION;
export const FOCUS_AREA_INFO = 'Focus Area Information';
export type FOCUS_AREA_INFO = typeof FOCUS_AREA_INFO;
export const FOCUS_INVESTIGATIONS = 'Focus Investigations';
export type FOCUS_INVESTIGATIONS = typeof FOCUS_INVESTIGATIONS;
export const MAP_ID = 'map-1';
export type MAP_ID = typeof MAP_ID;
export const GEOJSON = 'geojson';
export type GEOJSON = typeof GEOJSON;
export const STRINGIFIED_GEOJSON = 'stringified-geojson';
export type STRINGIFIED_GEOJSON = typeof STRINGIFIED_GEOJSON;
export const FOCUS_AREA_HEADER = 'Focus Area';
export type FOCUS_AREA_HEADER = typeof FOCUS_AREA_HEADER;
export const REASON_HEADER = 'Reason';
export type REASON_HEADER = typeof REASON_HEADER;
export const STATUS_HEADER = 'Status';
export type STATUS_HEADER = typeof STATUS_HEADER;
export const CASE_NOTIF_DATE_HEADER = 'Case Notif. Date';
export type CASE_NOTIF_DATE_HEADER = typeof CASE_NOTIF_DATE_HEADER;
export const CASE_CLASSIFICATION_HEADER = 'Case Class.';
export type CASE_CLASSIFICATION_HEADER = typeof CASE_CLASSIFICATION_HEADER;
export const DEFINITIONS = 'Definitions';
export type DEFINITIONS = typeof DEFINITIONS;
export const IN = 'in';
export type IN = typeof IN;
export const DISTRICT = 'District';
export type DISTRICT = typeof DISTRICT;
export const CANTON = 'Canton';
export type CANTON = typeof CANTON;
export const FI_REASON = 'FI Reason';
export type FI_REASON = typeof FI_REASON;
export const FI_STATUS = 'FI Status';
export type FI_STATUS = typeof FI_STATUS;
export const ACTIVE_INVESTIGATION = 'Active Investigation';
export type ACTIVE_INVESTIGATION = typeof ACTIVE_INVESTIGATION;
export const COMPLETE = 'Complete';
export type COMPLETE = typeof COMPLETE;
export const NO = 'No';
export type NO = typeof NO;
export const RESPONSE = 'Response';
export type RESPONSE = typeof RESPONSE;
export const MEASURE = 'Measure';
export type MEASURE = typeof MEASURE;
export const OF = 'of';
export type OF = typeof OF;
export const MARK_AS_COMPLETE = 'Mark as complete';
export type MARK_AS_COMPLETE = typeof MARK_AS_COMPLETE;
export const TARGET = 'Target';
export type TARGET = typeof TARGET;
export const POLYGON = 'Polygon';
export type POLYGON = typeof POLYGON;
export const MULTI_POLYGON = 'MultiPolygon';
export type MULTI_POLYGON = typeof MULTI_POLYGON;
export const POINT = 'Point';
export type POINT = typeof POINT;
export const FEATURE = 'Feature';
export type FEATURE = typeof FEATURE;
export const FEATURE_COLLECTION = 'FeatureCollection';
export type FEATURE_COLLECTION = typeof FEATURE_COLLECTION;
export const MAIN_PLAN = 'main-plan-layer';
export type MAIN_PLAN = typeof MAIN_PLAN;
export const APP = 'APP';
export type APP = typeof APP;
export const NO_GEOMETRIES_RESPONSE = 'Goals have no Geometries';
export type NO_GEOMETRIES_RESPONSE = typeof NO_GEOMETRIES_RESPONSE;

// internal urls
export const LOGIN_URL = '/login';
export type LOGIN_URL = typeof LOGIN_URL;
export const LOGOUT_URL = '/logout';
export type LOGOUT_URL = typeof LOGOUT_URL;
export const HOME_URL = '/';
export type HOME_URL = typeof HOME_URL;
export const IRS_URL = '/irs';
export type IRS_URL = typeof IRS_URL;
export const FI_URL = '/focus-investigation';
export type FI_URL = typeof FI_URL;
export const FI_SINGLE_URL = '/focus-investigation/view';
export type FI_SINGLE_URL = typeof FI_SINGLE_URL;
export const FI_SINGLE_MAP_URL = '/focus-investigation/map';
export type FI_SINGLE_MAP_URL = typeof FI_SINGLE_MAP_URL;
export const FI_HISTORICAL_URL = '/focus-investigation/historical';
export type FI_HISTORICAL_URL = typeof FI_HISTORICAL_URL;

// colors
export const GREEN = 'Green';
export type GREEN = typeof GREEN;
export const YELLOW = 'Yellow';
export type YELLOW = typeof YELLOW;
export const ORANGE = 'Orange';
export type ORANGE = typeof ORANGE;
export const RED = 'Red';
export type RED = typeof RED;

// task action codes
export const RACD_REGISTER_FAMILY_CODE = 'RACD Register Family';
export type RACD_REGISTER_FAMILY_CODE = typeof RACD_REGISTER_FAMILY_CODE;
export const CASE_CONFIRMATION_CODE = 'Case Confirmation';
export type CASE_CONFIRMATION_CODE = typeof CASE_CONFIRMATION_CODE;
export const MOSQUITO_COLLECTION_CODE = 'Mosquito Collection';
export type MOSQUITO_COLLECTION_CODE = typeof MOSQUITO_COLLECTION_CODE;
export const BEDNET_DISTRIBUTION_CODE = 'Bednet Distribution';
export type BEDNET_DISTRIBUTION_CODE = typeof BEDNET_DISTRIBUTION_CODE;
export const LARVAL_DIPPING_CODE = 'Larval Dipping';
export type LARVAL_DIPPING_CODE = typeof LARVAL_DIPPING_CODE;
export const BLOOD_SCREENING_CODE = 'Blood Screening';
export type BLOOD_SCREENING_CODE = typeof BLOOD_SCREENING_CODE;
export const IRS_CODE = 'IRS';
export type IRS = typeof IRS_CODE;
