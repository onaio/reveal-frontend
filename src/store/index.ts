import { store } from '@opensrp/store';

export default store;
