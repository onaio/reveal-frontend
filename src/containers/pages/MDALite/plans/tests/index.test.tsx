import { mount, shallow } from 'enzyme';
import flushPromises from 'flush-promises';
import { createBrowserHistory } from 'history';
import React from 'react';
import { act } from 'react-dom/test-utils';
import { Provider } from 'react-redux';
import { Router } from 'react-router';
import { REPORT_MDA_LITE_PLAN_URL } from '../../../../../constants';
import store from '../../../../../store';
import { MDALitePlans } from '../../../../../store/ducks/superset/MDALite/tests/fixtures';
import ConnectedMDALitePlansList, { MDALitePlansList } from '../index';

jest.mock('../../../../../configs/env');

const history = createBrowserHistory();

const props = {
  history,
  location: {
    hash: '',
    pathname: REPORT_MDA_LITE_PLAN_URL,
    search: '',
    state: undefined,
  },
  match: {
    isExact: true,
    params: {},
    path: `${REPORT_MDA_LITE_PLAN_URL}/`,
    url: `${REPORT_MDA_LITE_PLAN_URL}/`,
  },
};

describe('components/MDALiteReports/MDALitePlansList', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('renders without crashing', () => {
    shallow(
      <Router history={history}>
        <MDALitePlansList {...props} />
      </Router>
    );
  });

  it('renders plan definition list correctly', async () => {
    const service = jest.fn().mockImplementationOnce(() => Promise.resolve(MDALitePlans));
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMDALitePlansList {...props} service={service} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(wrapper.find('h3.page-title').text()).toEqual('MDA Lite Plans');
    wrapper
      .find('BreadcrumbItem li')
      .forEach((item, i) => expect(item.text()).toMatchSnapshot(`breadcrumb item-${i + 1}`));
    wrapper.find('.thead .tr .th').forEach((item, i) => {
      expect(item.text()).toMatchSnapshot(`table header-${i + 1}`);
    });
    expect(wrapper.find('.tbody .tr').length).toEqual(3);
    // have searchbar
    expect(wrapper.find('.search-input-wrapper').length).toEqual(1);
    // have top and bottom pagination
    expect(wrapper.find('.pagination').length).toEqual(2);
    // have height and columns filters
    expect(
      wrapper
        .find('.filter-bar-btns span')
        .at(0)
        .text()
    ).toEqual('Row Height');
    expect(
      wrapper
        .find('.filter-bar-btns span')
        .at(1)
        .text()
    ).toEqual('Customize Columns');

    expect(wrapper.find('GenericPlansList').length).toBe(1);
    const GenericPlansListProps = wrapper.find('GenericPlansList').props() as any;
    expect(GenericPlansListProps.plans).toEqual(MDALitePlans);
    expect(GenericPlansListProps.pageTitle).toEqual('MDA Lite Plans');
    expect(GenericPlansListProps.pageUrl).toEqual('/intervention/mda-lite/report');
    expect(GenericPlansListProps.supersetReportingSlice).toEqual('1');

    expect(wrapper.find('.page-title').text()).toEqual('MDA Lite Plans');

    wrapper.unmount();
  });

  it('handles search correctly', async () => {
    const service = jest.fn().mockImplementationOnce(() => Promise.resolve(MDALitePlans));
    props.location.search = '?title=kenya';
    const wrapper = mount(
      <Provider store={store}>
        <Router history={history}>
          <ConnectedMDALitePlansList {...props} service={service} />
        </Router>
      </Provider>
    );
    await act(async () => {
      await flushPromises();
    });
    wrapper.update();

    expect(wrapper.find('.tbody .tr').length).toEqual(1);
    expect(wrapper.find('.tbody .tr .td a').text()).toEqual('MDA-Lite kenya 2021-01-08');

    expect((wrapper.find('GenericPlansList').props() as any).plans).toEqual([MDALitePlans[1]]);
  });
});
