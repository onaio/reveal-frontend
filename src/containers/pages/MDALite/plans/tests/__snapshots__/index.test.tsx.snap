// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/MDALiteReports/MDALitePlansList handles search correctly: search results props 1`] = `
Array [
  Object {
    "jurisdiction_root_parent_ids": "[\\"bac41d87-b09d-4e76-be3b-3b6d07399891\\"]",
    "plan_date": 1610064000000,
    "plan_effective_period_end": 1640822400000,
    "plan_effective_period_start": 1610064000000,
    "plan_id": "6b485d4b-1043-40ed-ab8a-c64d24045ada",
    "plan_intervention_type": "MDA-Lite",
    "plan_name": "MDA-Lite-2021-01-08",
    "plan_status": "active",
    "plan_title": "MDA-Lite kenya 2021-01-08",
  },
]
`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: GenericPlansList props 1`] = `
Object {
  "fetchPlans": [Function],
  "history": Object {
    "action": "POP",
    "block": [Function],
    "createHref": [Function],
    "go": [Function],
    "goBack": [Function],
    "goForward": [Function],
    "length": 1,
    "listen": [Function],
    "location": Object {
      "hash": "",
      "pathname": "/",
      "search": "",
      "state": undefined,
    },
    "push": [Function],
    "replace": [Function],
  },
  "location": Object {
    "hash": "",
    "pathname": "/intervention/mda-lite/report",
    "search": "",
    "state": undefined,
  },
  "match": Object {
    "isExact": true,
    "params": Object {},
    "path": "/intervention/mda-lite/report/",
    "url": "/intervention/mda-lite/report/",
  },
  "pageTitle": "MDA Lite Plans",
  "pageUrl": "/intervention/mda-lite/report",
  "plans": Array [
    Object {
      "jurisdiction_root_parent_ids": "[\\"bac41d87-b09d-4e76-be3b-3b6d07399891\\"]",
      "plan_date": 1614643200000,
      "plan_effective_period_end": 1616371200000,
      "plan_effective_period_start": 1614643200000,
      "plan_id": "5e396185-6094-4817-9dd4-24bcbbc698b0",
      "plan_intervention_type": "MDA-Lite",
      "plan_name": "MDA-Lite-2021-03-02",
      "plan_status": "active",
      "plan_title": "MDA-Lite 2021-03-02",
    },
    Object {
      "jurisdiction_root_parent_ids": "[\\"bac41d87-b09d-4e76-be3b-3b6d07399891\\"]",
      "plan_date": 1610064000000,
      "plan_effective_period_end": 1640822400000,
      "plan_effective_period_start": 1610064000000,
      "plan_id": "6b485d4b-1043-40ed-ab8a-c64d24045ada",
      "plan_intervention_type": "MDA-Lite",
      "plan_name": "MDA-Lite-2021-01-08",
      "plan_status": "active",
      "plan_title": "MDA-Lite kenya 2021-01-08",
    },
    Object {
      "jurisdiction_root_parent_ids": "[\\"bac41d87-b09d-4e76-be3b-3b6d07399891\\"]",
      "plan_date": 1613520000000,
      "plan_effective_period_end": 1615248000000,
      "plan_effective_period_start": 1613520000000,
      "plan_id": "bbaf9674-e381-45f1-94d1-23bd373bc88e",
      "plan_intervention_type": "MDA-Lite",
      "plan_name": "MDA-Lite-2021-02-17",
      "plan_status": "active",
      "plan_title": "MDA-Lite Em test3 2021-02-17",
    },
  ],
  "service": [MockFunction] {
    "calls": Array [
      Array [
        "1",
      ],
    ],
    "results": Array [
      Object {
        "type": "return",
        "value": Promise {},
      },
    ],
  },
  "supersetReportingSlice": "1",
}
`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: breadcrumb item-1 1`] = `"Home"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: breadcrumb item-2 1`] = `"MDA Lite Plans"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: table header-1 1`] = `"Title"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: table header-2 1`] = `"Date Created"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: table header-3 1`] = `"Start Date"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: table header-4 1`] = `"End Date"`;

exports[`components/MDALiteReports/MDALitePlansList renders plan definition list correctly: table header-5 1`] = `"Status"`;
